src/shipping/views/PriceRatesUpdate/PriceRatesUpdate.tsx

```diff
@@ -248,6 +243,42 @@ export const PriceRatesUpdate: React.FC<PriceRatesUpdateProps> = ({
     reset();
   };
 
+  const onPostalCodeAssign = (rule: MinMax) => {
+    if (!originalCodes.length) {
+      setOriginalCodes([...rate.postalCodeRules]);
+    }
+    if (
+      rate.postalCodeRules.filter(
+        item => item.start === rule.min && item.end === rule.max
+      ).length > 0
+    ) {
+      closeModal();
+      return;
+    }
+    const newCode = {
+      __typename: undefined,
+      end: rule.max,
+      id: undefined,
+      inclusionType,
+      start: rule.min
+    };
+    rate.postalCodeRules.push(newCode);
+    closeModal();
+  };
+
+  const onPostalCodeUnassign = code => {
+    if (code.id !== undefined) {
+      setCodesToDelete([...codesToDelete, code.id]);
+      rate.postalCodeRules = rate.postalCodeRules.filter(
+        rule => rule.id !== code.id
+      );
+    } else {
+      rate.postalCodeRules = filterPostalCodes(rate.postalCodeRules, code);
```

Comment:
Same here, you're treating `rate` as a state but modifying it has no effect. The only reason you see an update in the UI is because you're calling setState functions below and triggering a rerender with a new value of `rate` defined in line 94